cmd/ovirt-populator/ovirt-populator.go

```diff
@@ -0,0 +1,252 @@
+	kind       = "OvirtVolumePopulator"
+	resource   = "ovirtvolumepopulators"
+)
+
+func main() {
+	var mode, engineUrl, secretName, diskID, fileName, crName, crNamespace, httpEndpoint, metricsPath, masterURL, kubeconfig, imageName, namespace string
+	var showVersion bool
+
+	// Main arg
+	flag.StringVar(&mode, "mode", "", "Mode to run in (controller, populate)")
+	// Populate args
+	flag.StringVar(&engineUrl, "engine-url", "", "ovirt-engine url (https//engine.fqdn)")
+	flag.StringVar(&secretName, "secret-name", "", "secret containing oVirt credentials")
+	flag.StringVar(&diskID, "disk-id", "", "ovirt-engine disk id")
+	flag.StringVar(&fileName, "file-name", "", "File name to populate")
+	flag.StringVar(&crName, "cr-name", "", "Custom Resource instance name")
+	flag.StringVar(&crNamespace, "cr-namespace", "", "Custom Resource instancce namespace")
+
+	// Controller args
+	if f := flag.Lookup("kubeconfig"); f != nil {
+		kubeconfig = f.Value.String()
+	} else {
+		flag.StringVar(&kubeconfig, "kubeconfig", "", "Path to a kubeconfig. Only required if out-of-cluster.")
+	}
+	flag.StringVar(&masterURL, "master", "", "The address of the Kubernetes API server. Overrides any value in kubeconfig. Only required if out-of-cluster.")
+	flag.StringVar(&imageName, "image-name", "", "Image to use for populating")
+	// Metrics args
+	flag.StringVar(&httpEndpoint, "http-endpoint", "", "The TCP network address where the HTTP server for diagnostics, including metrics and leader election health check, will listen (example: `:8080`). The default is empty string, which means the server is disabled.")
+	flag.StringVar(&metricsPath, "metrics-path", "/metrics", "The HTTP path where prometheus metrics will be exposed. Default is `/metrics`.")
+	// Other args
+	flag.BoolVar(&showVersion, "version", false, "display the version string")
+	flag.StringVar(&namespace, "namespace", "konveyor-forklift", "Namespace to deploy controller")
+	flag.Parse()
+
+	if showVersion {
+		fmt.Println(os.Args[0], version)
+		os.Exit(0)
+	}
+
+	switch mode {
+	case "controller":
+		var (
+			gk  = schema.GroupKind{Group: groupName, Kind: kind}
+			gvr = schema.GroupVersionResource{Group: groupName, Version: apiVersion, Resource: resource}
+		)
+		populator_machinery.RunController(masterURL, kubeconfig, imageName, httpEndpoint, metricsPath,
+			namespace, prefix, gk, gvr, mountPath, devicePath, getPopulatorPodArgs)
+	case "populate":
+		populate(masterURL, kubeconfig, crName, engineUrl, secretName, diskID, fileName, namespace, crNamespace)
+	default:
+		klog.Fatalf("Invalid mode: %s", mode)
+	}
+}
+
+type engineConfig struct {
+	URL      string
+	username string
+	password string
+	ca       string
+}
+
+type TransferProgress struct {
+	Transferred uint64  `json:"transferred"`
+	Description string  `json:"description"`
+	Size        *uint64 `json:"size,omitempty"`
+	Elapsed     float64 `json:"elapsed"`
+}
+
+func getSecret(secretName, engineURL, namespace string) engineConfig {
```

Comment:
it doesn't 'get' a secret but an engine config, right? how about `loadEngineConfig`?